E3SM-Project/E3SM
# Convert p3_init from f90 to CXX

This removes the last f90 from P3.

Change list:
* Move table read/write implementation to CXX in p3_init_impl.hpp
* p3_init now returns a P3LookupTables object with all the views inside containing the tables.
* Rename init_kokkos_tables to get_global_tables
* Rename init_kokkos_ice_lookup_tables to get_global_ice_lookup_tables
* Remove all f90 bridge code
* Move p3_tables_setup to tests dir, cleans up top p3 directory
* Fix p3_tables_setup implementation; it got broken in recent PRs but we didn't notice because it didn't get built by default
* Remove P3GlobalForFortran test struct, just call p3_init!
* Remove P3InitAP3Data test struct

Debatable change:
I really wanted to have `P3LookupTables lookup_tables` be static in p3_init so they would not have to be re-created on every p3_init call. Unfortunately, Kokkos does not allow static views to be active on program shutdown, since this occurs after Kokkos::finalize, so we would need add a p3_finalize method and ensure it was called at the end of p3. This seems burdensome and error prone, so I just decided to re-read all the tables every time p3_init is called.

There are some round-off level diffs in P3, so:
[non-BFB]
